pkg/cmd/list/cache_settings/cache_settings.go

```diff
@@ -0,0 +1,119 @@
+package cachesettings
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/color"
+	"go.uber.org/zap"
+
+	"github.com/MakeNowJust/heredoc"
+	table "github.com/MaxwelMazur/tablecli"
+	msg "github.com/aziontech/azion-cli/messages/list/cache_settings"
+	api "github.com/aziontech/azion-cli/pkg/api/edge_applications"
+	"github.com/aziontech/azion-cli/pkg/cmdutil"
+	"github.com/aziontech/azion-cli/pkg/contracts"
+	"github.com/aziontech/azion-cli/pkg/logger"
+	"github.com/aziontech/azion-cli/utils"
+	"github.com/spf13/cobra"
+)
+
+var edgeApplicationID int64
+
+func NewCmd(f *cmdutil.Factory) *cobra.Command {
+	opts := &contracts.ListOptions{}
+	cmd := &cobra.Command{
+		Use:           msg.Usage,
+		Short:         msg.ShortDescription,
+		Long:          msg.LongDescription,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		Example: heredoc.Doc(`
+		$ azion list cache-setting --application-id 16736354321
+		$ azion list cache-setting --application-id 16736354321 --details
+        `),
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !cmd.Flags().Changed("application-id") {
+				answer, err := utils.AskInput(msg.AskInputApplicationID)
+				if err != nil {
+					return err
+				}
+
+				num, err := strconv.ParseInt(answer, 10, 64)
+				if err != nil {
+					logger.Debug("Error while converting answer to int64", zap.Error(err))
+					return msg.ErrorConvertIdApplication
+				}
+
+				edgeApplicationID = num
+			}
+
+			var numberPage int64 = opts.Page
+			if !cmd.Flags().Changed("page") && !cmd.Flags().Changed("page-size") {
+				for {
```

Comment:
you can improve this logic by following the example https://github.com/aziontech/azion/blob/dev/pkg/cmd/list/domains/domains.go 👍🏽

Reply:
done!